nautobot/extras/utils.py

```diff
@@ -648,3 +650,62 @@ def bulk_delete_with_bulk_change_logging(qs, batch_size=1000):
         finally:
             change_context.defer_object_changes = False
             change_context.reset_deferred_object_changes()
+
+
+def create_schedule(schedule_data, job_kwargs, job_model, user, approval_required, task_queue=None):
+    """
+    This is an internal function to create a scheduled job from API data.
+    It has to handle both once-offs (i.e. of type TYPE_FUTURE) and interval
+    jobs.
+    """
+    from nautobot.extras.models.jobs import ScheduledJob  # avoid circular import
+
+    type_ = schedule_data["interval"]
+    if type_ == JobExecutionType.TYPE_IMMEDIATELY:
+        time = timezone.now()
+        name = schedule_data.get("name") or f"{job_model.name} - {time}"
+    elif type_ == JobExecutionType.TYPE_CUSTOM:
+        time = schedule_data.get("start_time")  # doing .get("key", "default") returns None instead of "default"
+        if time is None:
+            # "start_time" is checked against models.ScheduledJob.earliest_possible_time()
+            # which returns timezone.now() + timedelta(seconds=15)
+            time = timezone.now() + timedelta(seconds=20)
+        name = schedule_data["name"]
+    else:
+        time = schedule_data["start_time"]
+        name = schedule_data["name"]
```

Comment:
Let's add some error checking here - for example if `TYPE_IMMEDIATELY`, specifying any `start_time` should be an error, not silently ignored; a `crontab` should only be specified for `TYPE_CUSTOM`, etc.

Reply:
The serializer validation handles API checks. For the UI, errors are avoided as the form dynamically displays the appropriate fields based on the selected interval.